src/Evaluators.h

```diff
@@ -8,21 +8,95 @@
 #include <vector>
 // TODO: Better evaluation
 #include <random>
+#include <iostream>
 
 static std::random_device rd;
 static std::uniform_real_distribution<double> u_dist;
 
 using DS = DomineeringState;
 
+// TO DO: make helper methods.
+
 struct EvalTakeAway {
     double operator()(const DS& state) const {
-        return u_dist(rd);
+	int count = 0;
+
+//	Either implement so it counts how many reserved are for opponent
+//	or counts how many spaces it takes away from opponent
+	// Counts how many reserved are for opponent
+	 for (int i = 0; i < state.ROWS; i++){
+                for (int j = 0; j < state.COLS; j++){
+                        if (state.getCurPlayerSym() == 'B' && state.getCell(i,j) == state.EMPTYSYM){
```

Comment:
I will fix all those suggestions! Also, can you call methods from inside a struct method?

-----Original Message-----
From: "Naoki Mizuno" <notifications@github.com>
Sent: ‎10/‎26/‎2015 11:40 PM
To: "NigoroJr/uccineering" <uccineering@noreply.github.com>
Cc: "kojsmn" <kojsmn@miamioh.edu>
Subject: Re: [uccineering] Dev (#6)

In src/Evaluators.h:
>  struct EvalTakeAway {
>      double operator()(const DS& state) const {
> -        return u_dist(rd);
> +	int count = 0;
> +
> +//	Either implement so it counts how many reserved are for opponent
> +//	or counts how many spaces it takes away from opponent
> +	// Counts how many reserved are for opponent
> +	 for (int i = 0; i < state.ROWS; i++){
> +                for (int j = 0; j < state.COLS; j++){
> +                        if (state.getCurPlayerSym() == 'B' && state.getCell(i,j) == state.EMPTYSYM){
Use state.HOMESYM and state.AWAYSYM instead of hardcoded 'B' and 'W'.
—
Reply to this email directly or view it on GitHub.